discovery/docker_discovery.go

```diff
@@ -303,7 +316,8 @@ func (d *DockerDiscovery) manageConnection(quit chan bool) {
 	for {
 		// Is the client connected?
 		if client == nil || client.Ping() != nil {
-			log.Warn("Lost connection to Docker, re-connecting")
+			err := client.Ping()
```

Comment:
get the client.Ping error and log it

Reply:
Won't this extra `client.Ping()` panic if `client == nil`?

Reply:
good catch, I'll fix this